services/web/client/source/class/qxapp/dev/fake/srv/restapi/Authentication.js

```diff
@@ -0,0 +1,104 @@
+/* eslint no-warning-comments: "off" */
+qx.Class.define("qxapp.dev.fake.srv.restapi.Authentication", {
+  type: "static",
+
+  statics: {
+    REMEMBER: false,
+
+    mockData: [{
+      method: "GET",
+      url: "api/v1.0/token",
+      response: function(request) {
+        console.log("Received request:", request);
+
+        // Defaults unauthorized
+        let status = 401;
+        let headers = {
+          "Content-Type": "application/json"
+        };
+        let body = null;
+
+        const login = qxapp.dev.fake.srv.restapi.Authentication.decodeAuthHeader(request.requestHeaders);
+
+        const userId = qxapp.dev.fake.srv.restapi.Authentication.checkCredentials(login);
+        if (userId !== null) {
+          console.debug("User ", qxapp.dev.fake.srv.db.User.DUMMYNAMES[userId], "is logging in ...");
+          status = 200;
+          body = {
+            token: qxapp.dev.fake.srv.restapi.Authentication.createToken(userId)
+          };
+        }
+
+        request.respond(status, headers, qx.lang.Json.stringify(body));
+      }
+    }],
+
+    createToken: function(userId, expiration=24) {
+      return "this-is-a-dummy-token-that-expires-in-" + String(expiration) + "hours-for-" + String(userId);
+    },
+
+    getUserIdFromToken: function(token) {
+      if (token.startsWith("this-is-a-dummy-token-that-expires-in-")) {
+        var parts = token.split("-");
+        return parts.pop();
+      }
+      return null;
+    },
+
+    checkCredentials: function(login) {
+      var userId = qxapp.dev.fake.srv.db.User.DUMMYNAMES.findIndex(function(userName, userIndex) {
+        const user = qxapp.dev.fake.srv.db.User.getObject(userIndex);
+        return (login.email == user.email || login.email == user.username) && login.password == user.passwordHash;
+      });
+      return userId>=0? userId: null;
+    },
+
+    /**
+     * Gets {email, password} from header
+     * produced by qx.io.request.authentication.Basic
+    */
+    decodeAuthHeader: function(requestHeaders) {
+      var res = {
+        email: null,
+        password: null
+      };
+      var header = requestHeaders["Authorization"];
+
+      // Remove 'Basic $value'
+      var value = header.split(" ")[1];
+      // parse '$username : $password'
+      var pair = qx.util.Base64.decode(value).split(":");
+      res["email"] = pair[0];
```

Comment:
`res.email`

Reply:
what is the advantages vs ``res["email"]`` ?